cfpq_pyalgo/pygraphblas/grammars/wcnf.py

```diff
@@ -79,7 +79,7 @@ def contains(self, word: str):
         return self._cfg.contains(word)
 
     @classmethod
-    def from_text(cls, text: str, start_variable: Variable = Variable("S")):
+    def from_text(cls, text: str, start_variable: Variable = Variable("S")) -> "WCNF":
         cfg = CFG.from_text(text, start_variable)
         return cls(cfg)
```

Comment:
Why we want to return "WCNF" here?

Reply:
This is the 2"second constructor". This function creates an instance of WCNF from the grammar represented as text and returns it. So return type must be `WCNF` (but double quotes are required to use the class name in its definition)